dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/AbstractJdbcTableManager.java

```diff
@@ -184,8 +184,15 @@ public void removeUpdatedData( AnalyticsTableUpdateParams params, List<Analytics
     @Override
     public void createTable( AnalyticsTable table )
     {
-        createTempTable( table );
-        createTempTablePartitions( table );
+        if ( getPartitionColumn() != null )
```

Comment:
Checking for null in business logic does not lead to very descriptive/readable code, better to use descriptive method names or variables.